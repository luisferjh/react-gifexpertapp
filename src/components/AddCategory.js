import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handInputChange = (e) => {
        setInputValue(e.target.value)          
        
        console.log('handleInputChange llamado')
    }

    const handleSubmit = (e)=>{
        e.preventDefault();            
        if(inputValue.trim().length > 2)
        {
            setCategories(category => [inputValue, ...category])
            setInputValue('')
        }

    }

    return (
        <>
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handInputChange}
            />
        </form>
            
        </>
    )
}

AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
}
