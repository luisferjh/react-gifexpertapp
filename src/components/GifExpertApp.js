import React, {useState} from 'react'
import {AddCategory} from './AddCategory'
import {GifGrid} from './GifGrid'

export const GifExpertApp = () => {

   // const categories = ['One Puch', 'samurai X', 'dragon ball']
    const [categories, setCategories] = useState(['samurai X'])

    // const handleAdd = ()=>{
    //     console.log(categories)
    //     //setCategories([...categories, 'Hunter x Hunter']) // alternativa para no borrar el estado
    //     setCategories(cats =>  [...cats, 'Hunter x Hunter']) // alternativa para no borrar el estado
    // }


    console.log(categories)
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />        

            <ol>
               {
                   categories.map((category, index) => ( // los parentesis indican que se devuelve un objeto
                    <GifGrid 
                        key={category}
                        category={category} />
                   ))
               }
            </ol>
           
        </>
    )
}
