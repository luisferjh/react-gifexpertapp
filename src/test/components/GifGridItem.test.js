import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en <GifGridItem />', () => {

    const id = "eo8UrZwpvjPHi";
    const title = "rurouni kenshin GIF"
    const url = "https://media4.giphy.com/media/eo8UrZwpvjPHi/giphy.gif?cid=c7b9e7e9attbsd94pk06j0y6kh0q18zsbphpk33sm11b5qzz&rid=giphy.gif&ct=g"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe renderizar el componente correctamente', ()=>{       

        expect(wrapper).toMatchSnapshot()

    })

    test('debe de tener un pÃ¡rrafo con el title', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        //console.log(img.html())
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div')
       //animame_fadeIn
       const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)
    })
})