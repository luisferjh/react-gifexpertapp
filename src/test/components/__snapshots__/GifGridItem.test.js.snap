// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe renderizar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="rurouni kenshin GIF"
    src="https://media4.giphy.com/media/eo8UrZwpvjPHi/giphy.gif?cid=c7b9e7e9attbsd94pk06j0y6kh0q18zsbphpk33sm11b5qzz&rid=giphy.gif&ct=g"
  />
  <p>
    rurouni kenshin GIF
  </p>
</div>
`;
